### PR TITLE
Pract.07: Add 01 | Natalia Nakollofski

### PR DESCRIPTION
**Pull request by NataliaN24**
Natalia Nakollofski SI 1 1MI8000034

### CODE REVIEW

**Comment by Justsvetoslavov** [comment]
@GeorgiTerziev02 @khatoni Can you take a look?

File: Seminars/Sem.07/Pract. 07/Task01/String.h [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+	String& operator +=(const String& );
+	friend String& operator+(const String& lhs, const String& rhs);
+	friend bool operator ==(const String& lhs, const String& rhs);
+	friend bool operator !=(const String& lhs, const String& rhs);
+	friend bool operator <(const String& lhs, const String& rhs);
+	friend bool operator >(const String& lhs, const String& rhs);
+	friend bool operator <=(const String& lhs, const String& rhs);
+	friend bool operator >=(const String& lhs, const String& rhs);
+
+	size_t length()const;
```
Comment (GeorgiTerziev02) [inline Seminars/Sem.07/Pract. 07/Task01/String.h:30]: inconsistent with the private member

File: Seminars/Sem.07/Pract. 07/Task01/String.cpp [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+const char* String::c_str()const {
+	return content;
+}
+
+void String::setName(const char* str) {
+	delete[]content;
+	size = strlen(str);
+	content = new char[size + 1];
+	strcpy(content, str);
+}
```
Comment (GeorgiTerziev02) [inline Seminars/Sem.07/Pract. 07/Task01/String.cpp:59]: Why set name?

File: Seminars/Sem.07/Pract. 07/Task01/String.h [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+public:
+
+	String();
+	String(const char* );
+	String(const String&);
+	String& operator=(const String&);
+	~String();
+	
+	char operator[](size_t index)const;
```
Comment (GeorgiTerziev02) [inline Seminars/Sem.07/Pract. 07/Task01/String.h:18]: should have two funcs, one constant and one not

File: Seminars/Sem.07/Pract. 07/Task01/String.h [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+	friend bool operator ==(const String& lhs, const String& rhs);
+	friend bool operator !=(const String& lhs, const String& rhs);
+	friend bool operator <(const String& lhs, const String& rhs);
+	friend bool operator >(const String& lhs, const String& rhs);
+	friend bool operator <=(const String& lhs, const String& rhs);
+	friend bool operator >=(const String& lhs, const String& rhs);
+
+	size_t length()const;
+	const char* c_str()const;
+	void setName(const char*);
```
Comment (GeorgiTerziev02) [inline Seminars/Sem.07/Pract. 07/Task01/String.h:32]: ?

File: Seminars/Sem.07/Pract. 07/Task01/String.cpp [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+	
+	concat(other.content);
+	return *this;
+}
+
+bool   operator==(const String& lhs, const String& rhs) {
+	if (lhs.size != rhs.size) {
+		return false;
+	}
+	return( std::strcmp(lhs.content, rhs.content)==0);
```
Comment (GeorgiTerziev02) [inline Seminars/Sem.07/Pract. 07/Task01/String.cpp:87]: one line

Comment (GeorgiTerziev02) [reply Seminars/Sem.07/Pract. 07/Task01/String.cpp:87]: no need of the if

File: Seminars/Sem.07/Pract. 07/Task01/String.cpp [hunk 6]
```diff
@@ -0,0 +1,140 @@
...
+	char* newContent = new char[size + 2];
+	strcpy(newContent, content);
+	newContent[size] = ch;
+	newContent[size + 1] = '\0';
+	delete[] content;
+	content = newContent;
+}
+
+int String::compare(const String& other)const {
+	return strcmp(content, other.content);
```
Comment (GeorgiTerziev02) [inline Seminars/Sem.07/Pract. 07/Task01/String.cpp:139]: reuse oper==

File: Seminars/Sem.07/Pract. 07/Task01/String.cpp [hunk 7]
```diff
@@ -0,0 +1,140 @@
...
+	}
+	return( std::strcmp(lhs.content, rhs.content)==0);
+
+}
+
+bool operator !=(const String& lhs, const String& rhs) {
+	return !(lhs == rhs);
+}
+
+String& operator+(const String& lhs, const String& rhs) {
```
Comment (GeorgiTerziev02) [inline Seminars/Sem.07/Pract. 07/Task01/String.cpp:95]: Not by ref

File: Seminars/Sem.07/Pract. 07/Task01/String.h [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+	String(const char* );
+	String(const String&);
+	String& operator=(const String&);
+	~String();
+	
+	char operator[](size_t index)const;
+	void concat(const String& other);
+	
+	String& operator +=(const String& );
+	friend String& operator+(const String& lhs, const String& rhs);
```
Comment (GeorgiTerziev02) [inline Seminars/Sem.07/Pract. 07/Task01/String.h:22]: Do not return by ref

**Review by GeorgiTerziev02 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**